staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -264,6 +277,8 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 		s.GenericAPIServer.AuditBackend.Shutdown()
 	}
 
+	s.RunPreShutdownHooks()
+
 	return nil
```

Comment:
shouldn't we return the err of the hooks here?

Reply:
RunPreShutdownHooks currently doesn't return anything so there is no error to return.  The errors from each hook will be printed via runPreShutdownHook, so I'm not sure if there is much benefit to returning the errors from RunPreShutdownHooks.  Would we want preparedGenericAPIServer.Run or its caller to do something different if the shutdown hooks fail?  I was thinking the hooks run but if they fail or have issues the best they can do it print an error message.  We wouldn't want shutdown to be halted I wouldn't think.

Reply:
If audit shutdown gets broken, nobody will ever notice (via CI). I would prefer to fall over non-nil errors in integration tests.

Reply:
Ok, will re-work to pass the errors out